src/app/apartment/apartment-create/apartment-create.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { UserService } from '../../user/user.service';
+import { AuthenticationBasicService } from '../../login-basic/authentication-basic.service';
+import { Apartment } from '../apartment';
+import { Room } from '../../room/room';
+import { User } from '../../login-basic/user';
+import { FormsModule } from '@angular/forms';
+import { ErrorMessageService } from '../../error-handler/error-message.service';
+import { CommonModule } from '@angular/common';
+import { ApartmentService } from '../apartment.service';
+
+@Component({
+  selector: 'app-apartment-create',
+  standalone: true,
+  imports: [FormsModule, CommonModule],
+  templateUrl: './apartment-create.component.html',
+  styleUrls: ['./apartment-create.component.css']
+})
+export class ApartmentCreateComponent implements OnInit {
+  public apartment: Apartment = new Apartment();
+  public user: User = new User();
+  public isAuthorized: boolean = false;
+
+  constructor(
+    private router: Router,
+    private authenticationService: AuthenticationBasicService,
+    private userService: UserService,
+    private errorMessageService: ErrorMessageService,
+    private apartmentService: ApartmentService
+  ) {}
+
+  ngOnInit(): void {
+    this.user = this.authenticationService.getCurrentUser();
+    this.isAuthorized = this.isAuthorised();
+  }
+
+  onSubmit(): void {
+    if (!this.isAuthorized) {
+      this.onUnauthorised();
+      return;
+    }
+
+    this.apartment.owner = this.user;
+    this.apartment.registrationDate = new Date();
+
+    // TODO: Remove this Room mock
+    const room: Room = new Room({
+      surface: 0,
+      isOccupied: false,
+      hasWindow: false,
+      hasDesk: false,
+      hasBed: false,
+      ownerId: this.user.username
+    });
+    this.apartment.rooms = [room];
+
+    this.apartmentService.createResource({ body: this.apartment }).subscribe(
+      () => {
+        // TODO: Redirect to the apartment detail page
+        this.router.navigate(['/apartments']);
+      },
+      (error) => {
+        console.error('Error creating apartment:', error);
+        this.errorMessageService.showErrorMessage('Failed to create apartment. Please try again.');
+      }
+    );
+  }
+
+  private isAuthorised(): boolean {
+    return this.user.getRoles().includes('admin') || this.user.getRoles().includes('owner');
+  }
+
+  onUnauthorised(): void {
```

Comment:
It doesn't make sense to use onUnauthorised after submitting because, according to the ngIf condition, an unauthorized user will never be able to submit since the form is not displayed to them. You could include it in ngOnInit and add a sleep so the user can read the message before being redirected.

Reply:
Yeah, I thought about that but since user could be unauthorised by the ws by any reason that extra validation is not redundant imo.